userspace/libsinsp/parsers.cpp

```diff
@@ -1276,9 +1276,14 @@ void sinsp_parser::parse_clone_exit_caller(sinsp_evt *evt, int64_t child_tid) {
 		return;
 	}
 
-	/* If there's a listener, invoke it */
+	//
+	// If there's a listener, add a callback to later invoke it.
+	//
 	if(m_inspector->get_observer()) {
-		m_inspector->get_observer()->on_clone(evt, new_child.get(), tid_collision);
+		m_inspector->m_post_process_cbs.emplace(
+		        [&new_child, tid_collision](sinsp_observer *observer, sinsp_evt *evt) {
```

Comment:
Is there a reason for passing `new_child` by reference here? I see this being applied to other parameters below. I would need to double check, but I think we risk this pointing to potential garbage memory as `new_child` is a pointer variable on the stack, and the reference would get treated as a pointer to it. If it's about raw pointers, can't we just pass them by copy?

Reply:
You are right, great catch, damnit! 
/hold

Reply:
Could it be because `new_child` is already a reference to a `shared_ptr` (!) which is not owned by the caller and we don't want to increase the refcount in this case?

Reply:
I don't think that the added test covers all cases, thus i agree with Jason, we need the fix.
Also, i am not sure how to fix the `sinsp_parser::erase_fd()` case, since the function takes a stack pointer to an ad-hoc structure (https://github.com/falcosecurity/libs/pull/2222/files#diff-b1e2f413bae0df2c1d50354a44678c2350f5ce108a3609fe14896be2529873f8R3221) and, even when captured as a normal C pointer, its address will be lost once we call our callback.

Reply:
Ok i think i fixed everything; also i improved the new test so that it checks that listener params are ok.